crates/bls-gadgets/src/bls.rs

```diff
@@ -167,6 +168,34 @@ where
         Ok(prepared_aggregated_pk)
     }
 
+    /// Returns a gadget which checks that an aggregate pubkey is correctly calculated
+    /// by the sum of the pub keys
+    pub fn enforce_aggregated_all_pubkeys<CS: ConstraintSystem<F>>(
```

Comment:
To avoid code duplication, can you make the `enforce_aggregated_pubkeys` function call this, or remove `enforce_aggregated_pubkeys` and do the preparing in its callers?

Reply:
Not sure what you mean